server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java

```diff
@@ -277,11 +281,31 @@ private void updateBatchFetched(boolean success,
             }
         }
 
-        // schedule next poll
-        cancellable = threadPool.schedule(
-            this::requestBatchToFetch,
-            replicationSettings.pollDelay(),
-            ThreadPool.Names.LOGICAL_REPLICATION
+        // Renew retention lease with global checkpoint so that any shard that picks up shard replication task
```

Comment:
Shards also have local checkpoints, which describe the lag the shard. I guess global checkpoint is safe to use, but could that be less efficient (e.g. data processed twice) ?

Reply:
See https://github.com/opendistro-for-elasticsearch/cross-cluster-replication/blob/eebceb244d7456b2c675b8e6765f3cbccd316a9b/src/main/kotlin/com/amazon/elasticsearch/replication/task/shard/ShardReplicationTask.kt#L116

Reply:
Using the local checkpoint isn't safe to use here as it does not reflect the current "safe" state of all shards on the publishing cluster. Consider a failure on the primary shard with a local checkpoint 10, while the replica's local checkpoint and global checkpoint is still on 9. If the the local checkpoint 10 would be used and already replicated to any subscriber, but on the publisher side the repopulated (global checkpoint) is still on 9, an invariant state would have been replicated.

Opendistro's CCR seems to be outdated and afaik is superseded by OpenSearch's CCR which also (correctly) do not use an unsafe local checkpoint: https://github.com/opensearch-project/cross-cluster-replication/blob/main/src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt#L204

Reply:
👍

Reply:
This makes totally sense.